### PR TITLE
Safeguard `has_blocks()` against fatal errors.

### PR DESCRIPTION
**Pull request by costdev**
Trac ticket: https://core.trac.wordpress.org/ticket/55705

### CODE REVIEW

File: src/wp-includes/blocks.php [hunk 1]
```diff
@@ -425,6 +425,21 @@ function unregister_block_type( $name ) {
...
+		if ( null === $wp_post ) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %s: The $post variable. */
+					__( '%s is not a valid post.' ),
+					'<code>$post</code>'
+				),
+				'6.1.0'
+			);
```
Comment (dlh01) [inline src/wp-includes/blocks.php]: 👋 As an observer, `_doing_it_wrong()` seems a little strong. What's the thinking behind it? Most other core functions would just `return false` (or whatever appropriate value) so triggering a notice here seems inconsistent. Also, what's "wrong" with this case? The post might not exist, but is the function being used incorrectly?

Comment (costdev) [reply src/wp-includes/blocks.php]: Hi @dlh01, after an impromptu chat with @peterwilsoncc and reviewing other instances in Core, I believe you're absolutely right! Thanks! I'll make the changes to the PR now.

**Comment by audrasjb** [comment]
I tested the PR and it looks good to me 👍

**Review by audrasjb — APPROVED** [review]

**Review by dlh01 — COMMENTED** [review]
👍 LGTM!

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/53858

**Pull request closed without merging**